packages/dart_firebase_admin/lib/src/app/credential.dart

```diff
@@ -41,11 +43,25 @@ class Credential {
   }
 
   /// Log in to firebase using the environment variable.
-  Credential.fromApplicationDefaultCredentials({String? serviceAccountId})
-      : this._(
-          null,
-          serviceAccountId: serviceAccountId,
-        );
+  factory Credential.fromApplicationDefaultCredentials({
+    String? serviceAccountId,
+  }) {
+    ServiceAccountCredentials? creds;
+    final maybeConfig = platform.environment['GOOGLE_APPLICATION_CREDENTIALS'];
+    if (maybeConfig != null) {
+      try {
+        final decodedValue = jsonDecode(maybeConfig);
+        if (decodedValue is Map) {
+          creds = ServiceAccountCredentials.fromJson(decodedValue);
+        }
+      } catch (_) {}
```

Comment:
We shouldn't catch _anything_.
Only FormatExceptions.

Reply:
Resolved.